test/test_browser.py

```diff
@@ -5548,13 +5548,13 @@ def test_error_reporting(self):
   def test_webpack(self, es6):
     if es6:
       shutil.copytree(test_file('webpack_es6'), 'webpack')
-      self.emcc_args += ['-sEXPORT_ES6']
+      self.emcc_args += ['-sEXPORT_ES6', '-pthread', '-sPTHREAD_POOL_SIZE=1']
```

Comment:
I think it should be enough to instead use `-sPROXY_TO_PTHREAD`.    That way not only do we ensure that a thread is started but we actually run some code (the main function) on that thread.

Reply:
I tried `-sPROXY_TO_PTHREAD` initially but the test hangs instead of explicitly failing even though the same failure case occurs because the exception is caught by `callMain`: https://github.com/emscripten-core/emscripten/blob/34c1aa36052b1882058f22aa1916437ba0872690/src/postamble.js#L115 and doesn't seem to bubble up to where it'd crash/fail at the top level of the test JS code. If I build the test with `-sPTHREADS_DEBUG`, I can see it tries to load the worker from `file://`, but the exception doesn't make it up to where the test harness can see it failed.

Is there a way I could fix that in the test case to see the exception?

As an semi-related aside, in my own apps I've been building w/  `-sINVOKE_RUN=0` because `callMain` would gobble exceptions/crashes from Wasm making it hard to debug.

Reply:
What do you mean by "would gobble exceptions/crashes from Wasm"?   callMain should only deal with unwind/exit exceptions and should rethrow all other exceptions (i.e. exceptions that are not specifically supposed to be caught at the entry point)..  Are you not seeing that behaviour?   Or is it the rethrowning itself that is an issue?

Reply:
Yeah exactly, when I step through I see that in `handleException` it checks for those unwind/exit exceptions and then goes to `quit_` where it will rethrow the exception, but after that throw I just end up at this line: https://github.com/emscripten-core/emscripten/blob/main/test/webpack_es6/src/index.mjs#L16 where it prints loaded, and the exception seems to have disappeared? It's pretty strange, since the throw just seems to vanish after that point, when stepping through I don't end up at any other catch block/promise catch handling.

Reply:
Stepping a bit more, I see that it does end up catching the error when calling `instantiateAsync` and rejecting the promise, but that promise rejection doesn't seem to make it back up the app instantiating the module, so `index.mjs` doesn't see the rejection

Reply:
Well that certainly does seem like something worth investigating more.   Do you have a global unhandled rejection handler on your page?  Is your expectation that the module promise would be rejected?  I support that would be the most logical expectation.

Reply:
I don't have a global handler on my page, but the stepping through I'm describing here was actually just on running the test case: https://github.com/emscripten-core/emscripten/blob/main/test/webpack_es6/src/index.mjs . I think that the module promise being rejected if an error was thrown during callMain would be expected behavior, for example if the test code was like below, I'd expect the error to end up in the `catch` handler and print the error

```js
import Module from './hello.mjs';
Module(params).then((instance) => {
  console.log('loaded');
}).catch((error) => {
  console.error(`Error: ${error}`);
})
```

The `instantiateAsync` call: https://github.com/emscripten-core/emscripten/blob/34c1aa36052b1882058f22aa1916437ba0872690/src/preamble.js#L1082 has a catch handler that's fired after the exception is rethrown since that's the next spot that catches it, and it calls the promise reject function. However that promise rejection doesn't make it up to the top level of the app so the `Module(params)` promise is seen as successful instead of rejected (since `.then` is called instead)